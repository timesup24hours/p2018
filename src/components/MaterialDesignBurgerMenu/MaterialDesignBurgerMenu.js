import React from 'react';

import styles from './index.css';

// export default ({ CustomhandleOnClick }) => {
//   let spanRef;
//   const handleOnClick = e => {
//     if (CustomhandleOnClick) CustomhandleOnClick(e);
//     spanRef.classList.toggle(styles.trigger);
//   };
//   return (
//     <div className={styles.wrapper} onClick={handleOnClick}>
//       <div className={styles.spanWrapper}>
//         <span ref={ref => (spanRef = ref)} className={styles.stick} />
//       </div>
//     </div>
//   );
// };

export default class MaterialDesignBurgerMenu extends React.PureComponent {
  state = {
    open: false
  };

  handleOnClick = e => {
    if (this.props.CustomhandleOnClick) this.props.CustomhandleOnClick(e);
    this.spanRef.classList.toggle(styles.trigger);
  };

  // static getDerivedStateFromProps(props, state) {
  //   console.log(props, ' ', state);
  //   if (props.open !== state.open) {
  //     return {
  //       open: props.open
  //     };
  //   }
  //   return null;
  // }

  componentDidUpdate(prevProps, prevState, snapshot) {
    if (this.props.open) {
      this.spanRef.classList.add(styles.trigger);
    } else {
      this.spanRef.classList.remove(styles.trigger);
    }
    // if (!this.props.open && this.spanRef.classList.contains(styles.trigger)) {
    //   this.spanRef.classList.remove(styles.trigger);
    // } else if (
    //   this.props.open &&
    //   !this.spanRef.classList.contains(styles.trigger)
    // ) {
    //   this.spanRef.classList.add(styles.trigger);
    // }
  }

  render() {
    return (
      <div className={styles.wrapper} onClick={this.handleOnClick}>
        <div className={styles.spanWrapper}>
          <span ref={ref => (this.spanRef = ref)} className={styles.stick} />
        </div>
      </div>
    );
  }
}
