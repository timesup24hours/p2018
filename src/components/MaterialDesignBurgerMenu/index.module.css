.container {
  position: absolute;
  display: block;
  cursor: pointer;
  width: 40px;
  height: 40px;
}

.wrapper {
  height: 100%;
  width: 100%;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  /* background: white; */
  /* border-radius: 4px; */
  /* padding-bottom: 4px; */
  /* border: 1px solid black; */
}
.spanWrapper {
  height: 80%;
  width: 80%;
  position: relative;

  /* border: 1px solid darkgreen; */
  display: flex;
  justify-content: center;
  align-items: center;
}

.stick,
.stick:before,
.stick:after {
  width: 100%;
  height: 4px;
  /* margin: 0 auto; */
  /* background: #75489f; */
  background: white;
  position: absolute;
  transition: all 0.6s;
}

.stick:before {
  content: '';
  top: -13px;
  left: 0;
  position: absolute;
}

.stick:after {
  content: '';
  top: 13px;
  left: 0;
  position: absolute;
}

.stick.trigger {
  transform: rotate(180deg);
}

.stick.trigger:after {
  transform-origin: 100% 50%;
  transform: translateY(-14px) rotate(-45deg) scaleX(0.7);
}

.stick.trigger:before {
  transform-origin: 100% 50%;
  transform: translateY(14px) rotate(45deg) scaleX(0.7);
}
