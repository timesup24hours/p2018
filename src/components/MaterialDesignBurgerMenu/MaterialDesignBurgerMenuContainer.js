import React from 'react';

import styles from './index.module.css';
import MaterialDesignBurgerMenu from './MaterialDesignBurgerMenu';

export default () => {
  return (
    <div className={styles.container}>
      <MaterialDesignBurgerMenu />
    </div>
  );
};
