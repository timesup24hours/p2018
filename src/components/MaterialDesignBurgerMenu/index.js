// import React from 'react';
import MaterialDesignBurgerMenuContainer from './MaterialDesignBurgerMenuContainer';
import MaterialDesignBurgerMenu from './MaterialDesignBurgerMenu';

export { MaterialDesignBurgerMenuContainer, MaterialDesignBurgerMenu };
