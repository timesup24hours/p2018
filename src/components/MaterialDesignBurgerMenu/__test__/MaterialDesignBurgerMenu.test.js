import React from 'react';
import { shallow, mount } from 'enzyme';
import MaterialDesignBurgerMenu from '../MaterialDesignBurgerMenu';
import sinon from 'sinon';

describe('MobileMenuButton', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<MaterialDesignBurgerMenu debug />);

    expect(component).toMatchSnapshot();
  });

  it('passes function to the component', () => {
    const customhandleOnClick = jest.fn();
    const shallowWrapper = shallow(
      <MaterialDesignBurgerMenu customhandleOnClick={customhandleOnClick} />
    );
    const cmptProps = shallowWrapper.instance().props;

    expect(cmptProps.customhandleOnClick).toBe(customhandleOnClick);
  });

  it('onclick should work when passe function to the component', () => {
    const customhandleOnClick = jest.fn();
    const shallowWrapper = mount(
      <MaterialDesignBurgerMenu customhandleOnClick={customhandleOnClick} />
    );
    shallowWrapper.find('.wrapper').simulate('click');

    expect(customhandleOnClick).toBeCalledTimes(1);
  });

  it('onclick should have been called 2 times', () => {
    const customhandleOnClick = jest.fn();
    const shallowWrapper = mount(
      <MaterialDesignBurgerMenu customhandleOnClick={customhandleOnClick} />
    );
    shallowWrapper.find('.wrapper').simulate('click');
    shallowWrapper.find('.wrapper').simulate('click');

    expect(customhandleOnClick).toBeCalledTimes(2);
  });

  it('passes boolean to the component', () => {
    const open = true;
    const shallowWrapper = shallow(<MaterialDesignBurgerMenu open={open} />);
    const cmptProps = shallowWrapper.instance().props;

    expect(cmptProps.open).toBe(open);
  });

  it('componentDidUpdate should be called when click', () => {
    const customhandleOnClick = jest.fn();
    const component = mount(<MaterialDesignBurgerMenu />);

    const spy = sinon.spy(
      MaterialDesignBurgerMenu.prototype,
      'componentDidUpdate'
    );

    expect(spy).toHaveProperty('callCount', 0);

    component.setProps({ customhandleOnClick: customhandleOnClick });

    expect(spy).toHaveProperty('callCount', 1);

    component.setProps({ open: true });

    expect(spy).toHaveProperty('callCount', 2);
  });

  it('class length and class should be toggle when open props change ', () => {
    const component = mount(<MaterialDesignBurgerMenu />);

    const triggerClassName = component.find('.trigger');

    expect(triggerClassName).toHaveLength(0);
    // const spyPrev = jest.spyOn(component.instance().spanRef.classList, 'add');
    // expect(spyPrev).toHaveBeenCalled();

    expect(component.instance().spanRef.classList.length).toBe(1);
    expect(
      component.instance().spanRef.classList.contains('trigger')
    ).toBeFalsy();

    component.setProps({ open: true });

    expect(component.instance().spanRef.classList.length).toBe(2);
    expect(
      component.instance().spanRef.classList.contains('trigger')
    ).toBeTruthy();

    component.setProps({ open: false });

    expect(component.instance().spanRef.classList.length).toBe(1);
    expect(
      component.instance().spanRef.classList.contains('trigger')
    ).toBeFalsy();
  });

  it('class length and class should be toggle when props customhandleOnClick click  ', () => {
    const customhandleOnClick = jest.fn();
    const component = mount(<MaterialDesignBurgerMenu />);
    component.setProps({ customhandleOnClick: customhandleOnClick });

    expect(component.instance().spanRef.classList.length).toBe(1);
    expect(
      component.instance().spanRef.classList.contains('trigger')
    ).toBeFalsy();

    component.find('.wrapper').simulate('click');

    expect(component.instance().spanRef.classList.length).toBe(2);
    expect(
      component.instance().spanRef.classList.contains('trigger')
    ).toBeTruthy();
  });
});
