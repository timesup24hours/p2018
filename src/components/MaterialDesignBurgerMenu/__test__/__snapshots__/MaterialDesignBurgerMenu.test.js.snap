// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenuButton should render correctly in "debug" mode 1`] = `
<div
  className="wrapper"
  onClick={[Function]}
>
  <div
    className="spanWrapper"
  >
    <span
      className="stick"
    />
  </div>
</div>
`;
